src/NuGet.Core/NuGet.Protocol/Converters/PackageDependencyGroupConverter.cs

```diff
@@ -19,9 +17,31 @@ public class PackageDependencyGroupConverter : JsonConverter
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var set = JsonUtility.LoadJson(reader);
-
-            var fxName = set.Value<string>(JsonProperties.TargetFramework);
+            string fxName = null;
+            List<Packaging.Core.PackageDependency> packages = new List<Packaging.Core.PackageDependency>();
+            while (reader.Read() && reader.TokenType != JsonToken.EndObject)
+            {
+                if (reader.TokenType.Equals(JsonToken.PropertyName))
+                {
+                    if (reader.Value.Equals(JsonProperties.Dependencies))
+                    {
+                        // Dependencies are stored in an array
+                        while (reader.Read() && reader.TokenType != JsonToken.EndArray)
+                        {
+                            Packaging.Core.PackageDependency package = serializer.Deserialize<Packaging.Core.PackageDependency>(reader);
+                            packages.Add(package);
+                        }
```

Comment:
I replaced the above code with the following and the unit tests passed.

```cs
var packageDependencies = serializer.Deserialize<List<Packaging.Core.PackageDependency>>(reader);
packages.AddRange(packageDependencies);
```

I also removed the converter, and the tests passed locally.
```cs
        public static readonly JsonSerializerSettings ObjectSerializationSettings = new JsonSerializerSettings
        {
            MaxDepth = JsonSerializationMaxDepth,
            NullValueHandling = NullValueHandling.Ignore,
            TypeNameHandling = TypeNameHandling.None,
            Converters = new List<JsonConverter>
            {
                new NuGetVersionConverter(),
                new VersionInfoConverter(),
                new StringEnumConverter { NamingStrategy = new CamelCaseNamingStrategy() },
                new IsoDateTimeConverter { DateTimeStyles = DateTimeStyles.AssumeUniversal },
                new FingerprintsConverter(),
                new VersionRangeConverter(),
                new PackageVulnerabilityInfoConverter()
            },
        };
```

Reply:
I guess this is what Andy was referring in another comment is the converter really necesasary.

Reply:
Forgot to mention above, I also added `JsonConstructor` attribute to the `PackageDependency` class in `NuGet.Packaging` assembly for tests in `$\test\NuGet.Core.Tests\NuGet.Protocol.Tests\JsonExtensionsTest.cs` to pass. 

        [JsonConstructor]
        public PackageDependency(string id, VersionRange versionRange)
            : this(id, versionRange, include: null, exclude: null)
        {
        }

Reply:
Thanks for doing a deep dive into my code, I'll double check why mine wasn't working.

Reply:
It worked! The problem was that the `reader.Read();` was in the wrong place.

I'm decided to iterate over the json because it was throwing an error when opening the PM UI, it was complaining that I needed to save it to an JArray.